pymeasure/instruments/aja/dcxs.py

```diff
@@ -0,0 +1,229 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pyvisa
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+def truncated_string(value, maxlength):
+    """ Provides a validator function that truncates a string to a maximum length.
+    It returns the original value if its length is smaller or equal to the maximum,
+    otherwise only the characters up to the maximum length will be returned.
+
+    :param value: a value to test
+    :param maxlength: maximum length of the string ("values" arg of the property factories)
+    """
+    if len(value) <= maxlength:
+        return value
+    else:
+        return value[:maxlength]
+
+
+class DCXS(Instrument):
+    """ AJA DCXS-750 or 1500 DC magnetron sputtering power supply with multiple outputs
+
+    Connection to the device is made through an RS232 serial connection.
+    The communication settings are fixed in the device at 38400, one stopbit, no parity.
+    The communication protocol of the device uses single character commands and fixed length replys,
+    both without any terminator.
+
+    :param adapter: pyvisa resource name of the instrument or adapter instance
+    :param string name: The name of the instrument.
+    :param kwargs: Any valid key-word argument for Instrument
+    """
+
+    def __init__(self, adapter, name="AJA DCXS sputtering power supply", **kwargs):
+        kwargs.setdefault("asrl", dict(baud_rate=38400))
+        super().__init__(
+            adapter,
+            name,
+            includeSCPI=False,
+            write_termination="",
+            read_termination="",
+            **kwargs
+        )
+        # here we want to flush the read buffer since the device upon power up sends some '>'
+        # characters.
+        try:
+            self.adapter.flush_read_buffer()
+        except NotImplementedError:
+            # flush_read_buffer is not implemented for TCPIP sockets
+            try:
+                timeout = self.adapter.connection.timeout
+                self.adapter.connection.timeout = 0
+                try:
+                    self.read()
+                except pyvisa.errors.VisaIOError:
+                    # occurs always when calling read and no character is waiting
+                    pass
+                self.adapter.connection.timeout = timeout
+            except AttributeError:
+                # occurs in test suite (see #742 -> should be removed before merging)
+                pass
+        except AttributeError:
+            # occurs in test suite (see #742 -> should be removed before merging)
+            pass
```

Comment:
simplification after #836 

```suggestion
        self.adapter.flush_read_buffer()
```